pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -199,6 +199,8 @@ class PauliX(Observable, Operation):
 
     basis = "X"
 
+    batch_size = None
```

Comment:
What is up with all the `batch_size = None` - are they part of this PR or relics of other changes that made it into the branch? 😅

Reply:
Before I started just passing the pauli sentence on construction, one of the bottlenecks was calculating the batch size, since `SProd` was requesting it.  This made it just that a little bit quicker, since we know that non-parametric ops never have a batch size.

Reply:
Oh amazing! Hadnt even thought of that, great!